packages/itwin/grouping-mapping-widget/src/widget/components/customUI/GroupingMappingCustomUI.ts

```diff
@@ -16,20 +26,50 @@ export interface GroupingMappingCustomUI {
    * Display label in the widget.
    */
   displayLabel: string;
-  /**
-   * Custom UI Component to build query interactively. Refer to searchGroupingUI/manualGroupingUI.
-   */
-  uiComponent: (props: GroupingMappingCustomUIProps) => JSX.Element;
   /**
    * Optional icon, will be shown before display label in widget.
    */
   icon?: ReactElement;
 }
 
 /**
- * Customized UI component arguments definition.
+ * Context custom UI type definition.
+ */
+export interface ContextCustomUI extends IGroupingMappingCustomUI {
+  /**
+   * UI type.
+   */
+  type: GroupingMappingCustomUIType.CONTEXT;
+  /**
+   * User defined component for UI interaction with grouping mapping widget.
+   */
+  uiComponent?: () => JSX.Element;
+  /**
+   * Callback function for non UI interaction with grouping mapping widget.
+   */
+  callback?: (groupId: string, mappingId: string, iModelId: string) => void;
```

Comment:
The name `onClick` for this prop seems more appropriate then `callback` because this fires when you click on the custom context menu item.

Reply:
This is not a UI component prop. The uiComponent is a peer property in the same class so I think keep it as call back would be nice to match the requirement? But yes, for such external props, naming should be discussed. How do you think?@arnobmallickbsw

Reply:
I think `callback` is fine. It's more generic than `onClick`, this would still give us flexibility to change the actual event when this is fired. Maybe we will need to change it to some different event, e.g. `onChange` for some dropdown selection. Then we would need to rename this public property from `onClick` to `onChange`. I'd say keep the `callback` so that we do not need to change it later.

Reply:
IMO, I feel that it's too generic. It's not obvious right away when this actually gets called. If the behavior changes I believe it should be obvious and not keep the developer wondering why the behavior just randomly changed.